src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.X509.cs

```diff
@@ -32,6 +32,31 @@ internal static byte[] X509Encode(SafeX509Handle x)
 
             return encoded;
         }
+        [LibraryImport(Libraries.AndroidCryptoNative, EntryPoint = "AndroidCryptoNative_X509IsKeyStorePrivateKeyEntry")]
+        [return: MarshalAs(UnmanagedType.U1)]
```

Comment:
@simonrozsival the native code returns int32_t here so we should change this to UnmanagedType.Bool (4 bytes), or change the native code to use a C `bool` which is 1 byte. We seem to be using both variants in the Android crypto imports...

Reply:
At least in the Linux and MacOS Desktop PALs, we try to use `int32_t` and not a C `bool` since there is no guarantee by the C standard that it is exactly one byte.

Reply:
oh, thanks for pointing that out!

Reply:
https://github.com/dotnet/runtime/pull/103825

Reply:
@vcsjones there are plenty of examples of `UnmanagedType.Bool` even in the Unix PAL :smile: 

https://github.com/search?q=repo%3Adotnet%2Fruntime+UnmanagedType.Bool+path%3Asrc%2Flibraries%2FCommon%2Fsrc%2FInterop%2F+-path%3Asrc%2Flibraries%2FCommon%2Fsrc%2FInterop%2FWindows&type=code